### DISCUSSION TITLE
Lora

### DISCUSSION THREAD

**Discussion by utya** [arendst/Tasmota#12888] [Q&A] [answered]
Please, tell me if there is support for lora rfm96 / 98 modules?

**Comment by sfromis** [comment, marked as answer]
Not currently supported by Tasmota.

**Comment by barbudor** [comment]
What would be your use case?
Do you have sensors using Lora that you would like to receive in Tasmota?

**Reply by utya** [reply]
> What would be your use case?
> Do you have sensors using Lora that you would like to receive in Tasmota?
yes, i have

**Reply by barbudor** [reply]
As of today the easiest would be to use TasmotaClient:
Build a board with Arduino + your lora receiver.
Do all Lora processing in the Arduino and push the result to tasmota over serial using TasmotaClient library and Tasmota will adds the sensors from the Arduino into it's Telemetry messages.

**Reply by utya** [reply]
> As of today the easiest would be to use TasmotaClient:
> Build a board with Arduino + your lora receiver.
> Do all Lora processing in the Arduino and push the result to tasmota over serial using TasmotaClient library and Tasmota will adds the sensors from the Arduino into it's Telemetry messages.

thanks for advice

**Reply by c64emulator** [reply]
> What would be your use case? Do you have sensors using Lora that you would like to receive in Tasmota?

Here comes my use case for a ESP32 with LoRa (e.g. HELTEC LoRa32) on Tasmota:
I want to read the energy consumption in our house. In Germany (where we leave) the good old ferrari counters are replaced with new digital electricity meters. This meters have a IR interface with some specific protocol (SML).
The good news: Tasmota provides already support for this protocol. So far so good...
The bad news: This meters are placed somewhere in the house, where WiFi is not available (cellar, roof) or in metal cabinets where RF does not come through. Another use case: a friend is living in a neighborhood, where the meters of all houses are located in a central room, 200m from its own house.
Therefore I would give LoRa a chance.
My idea:
electricity meter with IR sender -> IR diode head -> ESP32 with Tasmota (for decoding) -> LoRA (for transmission) -> ESP32 with Tasmota (as receiver) -> MQTT

**Reply by barbudor** [reply]
Bad plan. Tasmota is meant to communicate with wifi using mqtt

The good plan would be
Sensor - > whatever CPU programmed in arduino /micropython - > Lora TX ~~~ Lora RX -> esp with tasmota - > MQTT/Wifi

**Reply by sfromis** [reply]
You want to place an ESP32 a place where RF does not come through, and then use LoRa.....

In general, I'd expect a high risk of placements for electricity meters being bad for weak radio signals like for LoRa. In a house, I'd prefer Ethernet if Wifi does not work well, and of course some cable length from the IR head to facilitate placement a bit away from the worst concentrations of metal.

**Reply by barbudor** [reply]
Still Lora can work across much longer distance than wifi. By using lower frequency and lower bit rate, range can be drastically higher than wifi.
I think Andreas Spies successfully tested a transmission around 200km
You also have Lora with low orbit satellites.

Of course, antenna should be outside the enclosure.

**Reply by joba-1** [reply]
> Of course, antenna should be outside the enclosure.

even that not really needed:

I watch the battery voltage of my car with LoRa.
The chip with an stm32 and including the 7cm wire antenna is inside the trunk. Car is in separate garage. I can receive the signal with an esp in the next house. I still use low range setting to save energy (2xAAA runs for ~2 years now…)

https://github.com/joba-1/LoraGw

https://github.com/joba-1/STM32L0LoraTx

![2C114ECD-EE42-4B9F-ACD0-B57B6F1D2F22](https://user-images.githubusercontent.com/32450554/196884012-5e9c9cc0-af87-4fa7-8438-47fdcf756e39.jpeg)

Only downside: no tasmota yet :)

**Reply by c64emulator** [reply]
> Bad plan. Tasmota is meant to communicate with wifi using mqtt
> 
> The good plan would be Sensor - > whatever CPU programmed in arduino /micropython - > Lora TX ~~~ Lora RX -> esp with tasmota - > MQTT/Wifi

I want to use Tasmota for the decoding of the SML protocol (Smart Meter Language). The implementation of SML decoding in Tasmota is straightforward and very flexible. I like it and would prefer usage. Could vanilla SML stream send via LoRa to ESP with Tasmota for decoding on reveiver side?

Good plan may be
Sensors (up to 4 IR heads) -> LoRa32  (ESP32 with LoRA, programmed in arduino/micropython) -> Lora TX ~~~ Lora RX -> LoRa32 with Tasmota (Tasmota for decoding the SML language) -> MQTT/Wifi

So Tasmota needs at least support for LoRa as receiver node?

**Reply by sfromis** [reply]
Are you coding then? :wink: 

Anyway, you'd also need a plan for making the SML decoding in Tasmota listen to the LoRa data stream. Of course, if you want to create LoRa support, it could also be based on transmitting the MQTT payloads to a receiver on the WLAN with MQTT.
